File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    let answer = []
    let kscore = []
    for (const s of score) {
        kscore.push(s)
        kscore.sort((a, b) => a - b)
        if (kscore.length > k) {
            kscore.shift()
        }
        answer.push(kscore[0])
    }
    return answer
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
import heapq

def solution(k, score):
    answer = []
    honor_k_heap = []
    for s in score:
        if len(honor_k_heap) < k:
            heapq.heappush(honor_k_heap, s)
        else:
            if s > honor_k_heap[0]:
                heapq.heapreplace(honor_k_heap, s)
        answer.append(honor_k_heap[0])
    return answer